cmd/juju/application/deployer/charm.go

```diff
@@ -526,6 +526,9 @@ func (c *repositoryCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI DeployerA
 		if termErr, ok := errors.Cause(err).(*common.TermsRequiredError); ok {
 			return errors.Trace(termErr.UserErr())
 		}
+		if origin.Source != commoncharm.OriginLocal && origin.Source != commoncharm.OriginCharmHub {
```

Comment:
We're making a huge assumption here about what the error is. If the facade call changes, it might not make sense any longer. Add some comments about that.

Reply:
Also, add into the error what the schema is, so that people will know. We don't document what a charm schema is what I found quickly.